activestorage/test/models/preview_test.rb

```diff
@@ -72,6 +72,27 @@ class ActiveStorage::PreviewTest < ActiveSupport::TestCase
     end
   end
 
+  test "image-related methods raise UnprocessedError when preview is not processed" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    preview = blob.preview(resize_to_limit: [640, 280])
+
+    assert_raises(ActiveStorage::Preview::UnprocessedError) { preview.url }
+    assert_raises(ActiveStorage::Preview::UnprocessedError) { preview.key }
+    assert_raises(ActiveStorage::Preview::UnprocessedError) { preview.download }
+  end
+
+  test "previewing with empty transformations does not generate a variant" do
+    blob = create_file_blob(filename: "report.pdf", content_type: "application/pdf")
+    preview = blob.preview({})
+
+    preview.processed
+
+    freeze_time { assert_equal blob.preview_image.url, preview.url }
```

Comment:
@chaadow I've stumbled upon this now, why is this here? If I remove it the tests still pass for me.